enhancements/upgrade-bundle.md

```diff
@@ -0,0 +1,260 @@
+2. Should we explicitly pass new version number as a flag or can we get it from bundle image?
+    * Based on discussions, can extract the version number from bundle image. 
+
+## Summary
+
+Enhance the Operator SDK to add a new subcommand `run bundle-upgrade` that performs an operator upgrade using OLM.
+
+## Motivation
+
+Testing operator upgrades remains difficult today due to the number of steps to prepare for an upgrade and the lack of tooling that automates those steps. Having high quality Operators is crucial to the success of the Operator Framework and thereby operator developers should have a tool to use to perform an operator upgrade.
+
+### Goals
+
+Provide a command line that upgrades an operator version using OLM.
+
+### Non-Goals
+
+This enhancement doesn't include the capability for SDK to use labels to detect the DB path. For the sake of simplicity, we will assume that the existing registry DB is located at `/database/index.db`.
+
+Another non-goal is that the enhancement is intended for operators that don't exist in the cluster. The scenario in which existing operators exist in the cluster is out of scope for this proposal.
+
+## Proposal
+
+### Implementation Details/Notes/Constraints [optional]
+
+The Operator SDK currently supports running an operator with OLM using the
+`run packagemanifests` subcommand. 
+This enhancement will add a new subcommand, `bundle-upgrade`,
+to the `run` command. This new command, `run bundle-upgrade`, will perform the upgrade of an operator from a previous version to a new specified version.
+
+Currently, the run command has different subcommands: 
+
+ * `run local`
+ * `run packagemanifests`
+ * `run bundle` - currently being implemented, design document is covered by [run bundle enhancement](https://github.com/operator-framework/enhancements/blob/master/enhancements/run-bundle.md)
+
+For doing an `bundle-upgrade`, there could be two scenarios: 
+
+* Case 1 - A previous version of the operator is already running on the cluster, and the index image contains this operator version. Now, we we want to upgrade the in cluster running operator to a new version that we specify. 
+
+* Case 2 - No previous version of an operator is running on the cluster, so we provide the new bundle image which is accessible and pullable within the cluster, which contains our desired version of the operator for the upgrade. 
```

Comment:
The way this is described, isn't it just install? or is the case of installing an initial version, and then upgrading to the target?

Reply:
This isn't just install. This command takes `--starting-csv` as a required flag, which specifies the first version of the CSV that OLM will install, and a positional argument `<bundle-image>` that specifies the newer version of the operator that we want to upgrade to.